libraries/WiFi/src/WiFiClient.cpp

```diff
@@ -224,6 +224,11 @@ size_t WiFiClient::write(const uint8_t *buf, size_t size)
     return res;
 }
 
+size_t WiFiClient::write_P(PGM_P buf, size_t size)
+{
+    return write(buf, size);
+}
+
```

Comment:
Instead of adding write_P perhaps adding macro defs to cores/esp32/pgmspace.h will work.
I looked around for other _P functions exported by ESP8266 core classes but I am not sure I got them all.

pgmspace.h
```C
#define write_P       write       /* WiFiClient::write_P, SdFile::write_P */
#define printf_P      printf      /* Print::printf_P */
#define writeln_P     writeln     /* SdFile::writeln_P */
```
This exactly equivalent to the ifdef but without developers having to worry about the issue. And without the need to add _P functions to ESP32 classes.